demos/nordic/nrf52840-dk/bootloader/main.c

```diff
@@ -273,29 +289,46 @@ int main( void )
         else
         {
             xErrCode = xVerifyImageSignature( ( uint8_t * ) CODE_REGION_2_START );
-
+            
             if( xErrCode != NRF_SUCCESS ) /* The firmware in the second bank has an
                                            * incorrect signature, erase its header */
             {
                 vEraseRegion( ( uint8_t * ) CODE_REGION_2_START, DESCRIPTOR_SIZE );
             }
             else
             {
+            /* We cannot boot from the second bank for now so we have to commit the image here. 
+            The bootloader can support booting from any position, but the firmware must be made relocatable
+            Unfortunately, it leaves out the case when developer distributed a bad firmware. 
+            
+            So, ideally, we need to get relocatable firmwares to work */
+
                 switch( BANK2_DESCRIPTOR->usImageFlags )
                 {
                     case IMAGE_FLAG_NEW:
```

Comment:
Appart from  vSetBankFlag(0, IMAGE_FLAG_COMMIT_PENDING ); This is a duplicated code from " case IMAGE_FLAG_VALID" can probably be coded in a more sustainable way.

Reply:
Done